src/JsonSchema/SchemaFactory.php

```diff
@@ -250,7 +250,12 @@ private function buildDefinitionName(string $className, string $format = 'json',
             $name = $groups ? sprintf('%s-%s', $prefix, implode('_', $groups)) : $prefix;
         }
 
-        return $name;
+        return $this->encodeDefinitionName($name);
+    }
+
+    private function encodeDefinitionName(string $name): string
+    {
+        return preg_replace('/^[a-zA-Z0-9.\-_]+$./', '.', $name);
```

Comment:
Your commit didn't work for me (tested with a "shortName" set to "directory/sub-directory" - still got the error "Could not resolve reference: #/components/schemas/..."), and I think the problem is the regular expression in that line. I changed it to `preg_replace('/[^a-zA-Z0-9.\-_]+/', '.', $name);` and then it worked for me - but I don't know if this is correct.

Reply:
@cziegenberg the correct regex should be `preg_replace('/[^a-zA-Z0-9.\-_]/', '.', $name)` in our case.
the additional `+` only says that the definition name must have 1 or more characters. But our use case is only to place INVALID characters.

Reply:
@cziegenberg Should be fixed. Can you test?

@dunglas Please review.

Reply:
@stchr I couldn't test it completely, because of other problems with the master branch, but the aforementioned error message doesn't occur anymore.